File: src/main/scala/im/tox/toktok/app/domain/Message.scala
package im.tox.toktok.app.domain

sealed abstract class MessageType(val viewType: Int)
object MessageType {
  case object Delivered extends MessageType(1)
  case object Received extends MessageType(2)
  case object Action extends MessageType(3)
}

final case class Message(
  msgType: MessageType,
  msgContent: String,
  msgDetails: String,
  imageSrc: Int
)

File: sources/app/src/main/java/im/tox/toktok/app/domain/Message.java
package im.tox.toktok.app.domain;

public class Message {
    public final MessageType msgType;
    public final String msgContent;
    public final String msgDetails;
    public final int imageSrc;

    public Message(
            MessageType msgType,
            String msgContent,
            String msgDetails,
            int imageSrc
    ) {
        this.msgType = msgType;
        this.msgContent = msgContent;
        this.msgDetails = msgDetails;
        this.imageSrc = imageSrc;
    }
}
